### PR TITLE
add some devdocs for `new_opaque_closure`

### PR DESCRIPTION
**Pull request by simeonschaub**

### CODE REVIEW

File: doc/src/devdocs/ast.md [hunk 1]
```diff
@@ -520,6 +520,36 @@ These symbols appear in the `head` field of [`Expr`](@ref)s in lowered form.
...
+
+        Indicates whether the closure accepts varargs.
+        
+      * `args[3]` : lb
+
+        Lower bound on the output type. (Defaults to `Union{}`)
+
+      * `args[4]` : ub
+
+        Lower bound on the output type. (Defaults to `Any`)
```
Comment (nickrobinson251) [inline doc/src/devdocs/ast.md:541]:
```suggestion
        Upper bound on the output type. (Defaults to `Any`)
```

Comment (simeonschaub) [reply doc/src/devdocs/ast.md]: Oops, thanks!

**Review by Keno — APPROVED** [review]
Probably needs much more explanation on what the semantics actually are, but this seems correct and least and incomplete is better than nothing.

**Comment by simeonschaub** [comment]
Yeah, I just wanted to write this down somewhere for myself, because I needed to look this up again.

**Pull request merged**